NEWS.md

```diff
@@ -1,5 +1,10 @@
 # tidyr (development version)
 
+* `unpack()` does a better job of reporting column name duplication issues and
+  gives better advise about how to resolve them using `names_sep`. This also
+  improves errors from functions that use `unpack()`, like `unnest()` and
+  `unnest_wider()` (#1425, #1367).
```

Comment:
I also checked what happens if you unnest a data frame with bad names:

```R
df <- tibble(
  x = list(data.frame(a = 1, a = 2, a = 3, check.names = FALSE))
)
df |> unnest(x)
```

And the error looks reasonable there too.

Reply:
There is actually a test for that case https://github.com/tidyverse/tidyr/pull/1448/files#diff-2d1025444fee4618ff7bfdb6355745f5664188d6133df3abf807c4dc2413cc7aR152-R161